benchmark/scripts/Benchmark_Driver

```diff
@@ -487,16 +497,14 @@ class BenchmarkDoctor(object):
         name = measurements["name"]
         setup, ratio = BenchmarkDoctor._setup_overhead(measurements)
         setup = 0 if ratio < 0.05 else setup
-        runtime = min(
-            [
-                (result.samples.min - correction)
-                for i_series in [
-                    BenchmarkDoctor._select(measurements, num_iters=i)
-                    for correction in [(setup / i) for i in [1, 2]]
-                ]
-                for result in i_series
-            ]
-        )
+
+        runtimes = []
+        for i in range(1, 3):
+            correction = setup / i
+            i_series = BenchmarkDoctor._select(measurements, num_iters=i)
+            for result in i_series:
+                runtimes.append(result.samples.min - correction)
+        runtime = min(runtimes)
```

Comment:
I'd like whoever will review this to pay attention at this particular change.

The old version relied on Python 2-only feature where the variables defined in a list comprehension leak to the function scope. But I believe this was erroneous: since in Python 2 list comprehensions are eagerly evaluated, the value of `i` was always 2 when evaluating the `BenchmarkDoctor._select(measurements, num_iters=i)` expression. Similarly, when evaluating `(result.samples.min - correction)`, the value of `correction` was always the same (namely, the last value it took while evaluating this expression:

```python
[
    BenchmarkDoctor._select(measurements, num_iters=i)
    for correction in [(setup / i) for i in [1, 2]]
]
```

Here, I've rewritten it to be a stupid old for loop, which IMHO reads way better, and also seems to behave correctly.

Reply:
Hmmm… I’ll need to check this.

Reply:
I’ve corrected the Python3 incompatibility by explicitly propagating the variables from the nested scopes. I don’t see any issue of incorrect evaluation. Here’s standalone simplified test case base on the original source code before the automatic formatter made mess of it in  #29719:

````python
$ python
Python 2.7.10 (default, Feb 22 2019, 21:55:15) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.37.14)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> setup = 134
>>> measurements = {1: [[324], [326], [325]], 2: [[257], [259], [258]]}
>>> 
>>> 
>>> def select(measurements, num_iters):
...     return measurements[num_iters]
... 
>>> 
>>> runtimesP3 = [
...     (result[0] - correction) for correction, i_series in
...     [(correction, select(measurements, num_iters=i))
...      for i, correction in [(i, setup // i) for i in [1, 2]]
...      ] for result in i_series]
>>> 
>>> runtimes = [
...     (result[0] - correction) for i_series in
...     [select(measurements, num_iters=i)
...      for correction in [(setup / i) for i in [1, 2]]
...      ] for result in i_series]
>>> 
>>> runtimes == runtimesP3
True
>>> runtimesP3
[190, 192, 191, 190, 192, 191]
````

It works correctly in Python 3, too:

````python
$ python3
Python 3.6.5 (default, May 24 2018, 10:57:11) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> setup = 134
>>> measurements = {1: [[324], [326], [325]], 2: [[257], [259], [258]]}
>>> 
>>> 
>>> def select(measurements, num_iters):
...     return measurements[num_iters]
... 
>>> 
>>> runtimesP3 = [
...     (result[0] - correction) for correction, i_series in
...     [(correction, select(measurements, num_iters=i))
...      for i, correction in [(i, setup // i) for i in [1, 2]]
...      ] for result in i_series]
>>> 
>>> runtimesP3
[190, 192, 191, 190, 192, 191]
````

I’ll try to incorporate the Python 3 changes in #26462 (it already had fix for the `assertEqualtats` typo...